cmd/kube-controller-manager/app/core.go

```diff
@@ -416,17 +416,22 @@ func startPVProtectionController(ctx ControllerContext) (bool, error) {
 
 func startSnapshotController(ctx ControllerContext) (bool, error) {
 	if utilfeature.DefaultFeatureGate.Enabled(features.VolumeSnapshot) {
-		ssc, err := snapshot.NewSnapshotController(
-			ctx.InformerFactory.Storage().V1alpha1().VolumeSnapshots(),
-			ctx.InformerFactory.Storage().V1alpha1().VolumeSnapshotDatas(),
-			ctx.ClientBuilder.ClientOrDie("snapshot-controller"),
-			ctx.Cloud,
-			ProbeSnapshottableVolumePlugins(ctx.ComponentConfig.PersistentVolumeBinderController.VolumeConfiguration),
-		)
+		params := snapshotcontroller.ControllerParameters{
+			KubeClient:                 ctx.ClientBuilder.ClientOrDie("snapshot-controller"),
+			SyncPeriod:                 ctx.ComponentConfig.PersistentVolumeBinderController.PVClaimBinderSyncPeriod.Duration,
+			VolumePlugins:              ProbeSnapshottableVolumePlugins(ctx.ComponentConfig.PersistentVolumeBinderController.VolumeConfiguration),
+			Cloud:                      ctx.Cloud,
+			ClusterName:                ctx.ComponentConfig.KubeCloudShared.ClusterName,
+			VolumeInformer:             ctx.InformerFactory.Core().V1().PersistentVolumes(),
+			ClaimInformer:              ctx.InformerFactory.Core().V1().PersistentVolumeClaims(),
+			VolumeSnapshotInformer:     ctx.InformerFactory.Storage().V1alpha1().VolumeSnapshots(),
+			VolumeSnapshotDataInformer: ctx.InformerFactory.Storage().V1alpha1().VolumeSnapshotDatas(),
+		}
+		ssc, err := snapshotcontroller.NewSnapshotController(params)
 		if err != nil {
-			return true, fmt.Errorf("Failed to start volume snapshot controller : %v", err)
+			return true, fmt.Errorf("failed to start volume snapshot controller : %v", err)
```

Comment:
Is it a convention to use lower case letter to start a message?

Reply:
Yes, `error strings should not be capitalized or end with punctuation`,  else there will be a weak warning.